cmd/archive_upload_client/client.go

```diff
@@ -62,8 +63,16 @@ func upload(ctx context.Context, conn *grpc.ClientConn, p *pb.FileRequest, audie
 	// A given TokenSource is specific to the audience.
 	tokenSource, err := idtoken.NewTokenSource(ctx, audience)
 	if err != nil {
-		return nil, fmt.Errorf("idtoken.NewTokenSource: %v", err)
+		if err.Error() != `idtoken: credential must be service_account, found "authorized_user"` {
+			return nil, fmt.Errorf("idtoken.NewTokenSource: %v", err)
+		}
```

Comment:
Why do we need this? Can we just remove this part?

Reply:
I think the point of this was to just validate: "did you get the error which can be worked around with this? or something different?"

ToTT says: "never depend on error text" so yea this is 'not terrific'... again, let's just toss this PR and move along with yours :)